openff/bespokefit/utilities/tempcd.py

```diff
@@ -0,0 +1,55 @@
+""""""
+import os
+import shutil
+import tempfile
+from contextlib import contextmanager
+from pathlib import Path
+from typing import Optional, Union
+
+from openff.bespokefit.executor.services import current_settings
+
+
+@contextmanager
+def temporary_cd(path: Optional[Union[str, Path]] = None):
+    """
+    Context manager to move the current working directory to the path specified.
+
+    If no path is given or the path does not exist, a temporary directory will
+    be created. This temporary directory and its contents will be deleted when
+    the context manager exits.
+
+    Parameters
+    ----------
+    path
+        The path to CD into. If ``None`` or not specified, a temporary directory
+        will be created. If specified but the path does not exist, a temporary
+        directory with that name will be created.
+    """
+    # Normalize path to a pathlib Path
+    path: Optional[Path] = None if path is None else Path(path)
+
+    # Decide whether to clean up based on bespokefit settings
+    cleanup = not current_settings().BEFLOW_KEEP_TMP_FILES
```

Comment:
This should depend on both the old and new variables.

Reply:
I wanted to keep the new behavior and the old behavior separate for the same breaking things reasons as above... in 0.3.0 or whatever we'll deprecate/remove the old setting altogether, but for now I don't wanna break any scripts people have that assume the files are going to a particular place.

Reply:
Welp ok yep so these changes break the old setting anyway - since the source and destination for the files is the same, the optimizer worker cleans up the copied files [here](https://github.com/openforcefield/openff-bespokefit/blob/6b57aa6464879a0cebc58b87f227b613e08d4fdd/openff/bespokefit/executor/executor.py#L347-L348). So forget everything I said your way is perfect :)